### ISSUE TITLE
mkfs.f2fs outdated, default command to convert /data to F2FS fails

### ISSUE THREAD

**Issue by issmirnov** [TeamWin/android_device_motorola_victara#2]
tl;dr - Decryption fails, formatting /data to F2FS impossible due to old mkfs.f2fs binary. Please pull upstream - affects all version of TWRP for victara.

---

Hello,

I recently needed to decrypt my XT1095. I rebooted into TWRP and ran "factory reset", and then attempted to manually format the data partition. Unfortunately, I kept getting the following error:

```
mkfs.f2fs -t 0 -r 16384  /dev/block/mmcblk0p39
process ended with ERROR: 1
Unable to wipe data
Error changing file system
```

Upon further examination, I was able to invoke the command through adb shell. It turns out the binary is version 1.1.0 (2012-11-29), and has the following usage:

```
/sbin # mkfs.f2fs 

    F2FS-tools: mkfs.f2fs Ver: 1.1.0 (2012-11-29)

    Error: Device not specified

Usage: mkfs.f2fs [options] device
[options]:
  -a heap-based allocation [default:1]
  -d debug level [default:0]
  -e [extension list] e.g. "mp3,gif,mov"
  -l label
  -o overprovision ratio [default:5]
  -s # of segments per section [default:1]
  -z # of sections per zone [default:1]
  -t 0: nodiscard, 1: discard [default:1]
```

As you can see, the `-r` flag is missing. 

I was ultimately able to resolve this issue by downloading the `twrp-2.8.7.1-hammerhead.img` image and using Android Image Kitchen to pull a newer binary from the ram disk, however this cost me a fair but of time.

My suggestion - could you look at https://github.com/CyanogenMod/android_external_f2fs-tools and merge in a newer release? As it stands now this particular build of TWRP will not be able to convert any partitions back to F2FS if they get formatted to ext4 during encryption.

**Comment by muralivijay** [comment]
same also muy device how to fix it ?
